trillian/ctfe/cert_checker_test.go

```diff
@@ -320,6 +321,77 @@ func TestNotAfterRange(t *testing.T) {
 	}
 }
 
+func TestRejectExpiredNonExpired(t *testing.T) {
+	fakeCARoots := NewPEMCertPool()
+	if !fakeCARoots.AppendCertsFromPEM([]byte(testonly.FakeCACertPEM)) {
+		t.Fatal("failed to load fake root")
+	}
+	chain := pemsToDERChain(t, []string{testonly.LeafSignedByFakeIntermediateCertPEM, testonly.FakeIntermediateCertPEM})
+	validateOpts := CertValidationOpts{
+		trustedRoots: fakeCARoots,
+		extKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
+	}
+
+	for _, tc := range []struct {
+		desc             string
+		rejectExpired    bool
+		rejectNonExpired bool
+		now              time.Time
+		wantErr          string
+	}{
+		{desc: "no-reject"},
+		{
+			desc:          "reject-expired-pass",
+			rejectExpired: true,
+			now:           time.Date(2017, 1, 1, 0, 0, 0, 0, time.UTC),
+		},
+		{
+			desc:          "reject-expired-after",
+			rejectExpired: true,
+			now:           time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+			wantErr:       "expired or is not yet valid",
+		},
+		{
+			desc:          "reject-expired-before",
+			rejectExpired: true,
+			now:           time.Date(1999, 1, 1, 0, 0, 0, 0, time.UTC),
+			wantErr:       "expired or is not yet valid",
+		},
+		{
+			desc:             "reject-non-expired-pass-after",
+			rejectNonExpired: true,
+			now:              time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC),
+		},
+		{
+			desc:             "reject-non-expired-pass-before",
+			rejectNonExpired: true,
+			now:              time.Date(1999, 1, 1, 0, 0, 0, 0, time.UTC),
+		},
+		{
+			desc:             "reject-non-expired",
+			rejectNonExpired: true,
+			now:              time.Date(2017, 1, 1, 0, 0, 0, 0, time.UTC),
+			wantErr:          "only expired certificates",
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			validateOpts.currentTime = tc.now
+			validateOpts.rejectExpired = tc.rejectExpired
+			validateOpts.rejectNonExpired = tc.rejectNonExpired
+			chainX509, err := ValidateChain(chain, validateOpts)
+			if err != nil {
+				if len(tc.wantErr) == 0 {
+					t.Errorf("ValidateChain()=_,%v; want _,nil", err)
+				} else if !strings.Contains(err.Error(), tc.wantErr) {
+					t.Errorf("ValidateChain()=_,%v; want err containing %q", err, tc.wantErr)
+				}
+			} else if len(tc.wantErr) != 0 {
+				t.Errorf("ValidateChain()=%v,nil; want err containing %q", chainX509, tc.wantErr)
```

Comment:
(nit) Suspect that printing out `chainX509` is going to be unhelpful if this test case ever fails.

Reply:
Removed.